components/core/src/clp/CurlDownloadHandler.hpp

```diff
@@ -91,6 +92,27 @@ class CurlDownloadHandler {
     [[nodiscard]] auto perform() -> CURLcode { return m_easy_handle.perform(); }
 
 private:
+    /**
+     * Locate the certificate authority(CA) bundle file on the current host.
+     *
+     * This provides consistent CA bundle resolution for curl on Linux systems when the runtime
+     * environment differs from the build environment. MacOS and Windows rely on their
+     * native certificate stores and are not checked. See also: https://curl.se/docs/sslcerts.html
+     *
+     * Resolution order:
+     *   1. Environment variables CURL_CA_BUNDLE then SSL_CERT_FILE, if set.
+     *   2. Well-known CA bundle paths.
+     *   3. If nothing is found, return an empty string. The caller needs to check the output so as
+     *      not to override curl defaults and let it use the platform's own certificate store or
+     *      whatever it was built with.
+     *
+     * @return Absolute path to the CA bundle file.
+     */
+    static auto get_host_ca_bundle_path() -> std::string;
```

Comment:
```suggestion
    [[nodiscard]] static auto get_host_ca_bundle_path() -> std::optional<std::string>;
```
Let's return `std::optional` instead.